src/input/input_handler.cpp

```diff
@@ -0,0 +1,694 @@
+        InputBinding binding = GetBindingFromString(input_string);
+        BindingConnection connection(0, nullptr);
+        auto button_it = string_to_cbutton_map.find(output_string);
+        auto axis_it = string_to_axis_map.find(output_string);
+
+        if (binding.IsEmpty()) {
+            LOG_WARNING(Input, "Invalid format at line: {}, data: \"{}\", skipping line.",
+                        lineCount, line);
+            continue;
+        }
+        if (button_it != string_to_cbutton_map.end()) {
+            connection =
+                BindingConnection(binding, GetOutputPointer(ControllerOutput(button_it->second)));
+            connections.insert(connections.end(), connection);
+
+        } else if (axis_it != string_to_axis_map.end()) {
+            int value_to_set = (binding.key3 & 0x80000000) != 0 ? 0
+                               : (axis_it->second.axis == Axis::TriggerLeft ||
+                                  axis_it->second.axis == Axis::TriggerRight)
+                                   ? 127
+                                   : axis_it->second.value;
+            connection =
+                BindingConnection(binding,
+                                  GetOutputPointer(ControllerOutput(OrbisPadButtonDataOffset::None,
+                                                                    axis_it->second.axis)),
+                                  value_to_set);
+            connections.insert(connections.end(), connection);
+        } else {
+            LOG_WARNING(Input, "Invalid format at line: {}, data: \"{}\", skipping line.",
+                        lineCount, line);
+            continue;
+        }
+        // LOG_INFO(Input, "Succesfully parsed line {}", lineCount);
+    }
+    file.close();
+    connections.sort();
+    LOG_DEBUG(Input, "Done parsing the input config!");
+}
+
+u32 GetMouseWheelEvent(const SDL_Event& event) {
+    if (event.type != SDL_EVENT_MOUSE_WHEEL && event.type != SDL_EVENT_MOUSE_WHEEL_OFF) {
+        LOG_DEBUG(Input, "Something went wrong with wheel input parsing!");
+        return (u32)-1;
+    }
+    if (event.wheel.y > 0) {
+        return SDL_MOUSE_WHEEL_UP;
+    } else if (event.wheel.y < 0) {
+        return SDL_MOUSE_WHEEL_DOWN;
+    } else if (event.wheel.x > 0) {
+        return SDL_MOUSE_WHEEL_RIGHT;
+    } else if (event.wheel.x < 0) {
+        return SDL_MOUSE_WHEEL_LEFT;
+    }
+    return (u32)-1;
+}
+
+u32 InputBinding::GetInputIDFromEvent(const SDL_Event& e) {
+    int value_mask;
+    switch (e.type) {
+    case SDL_EVENT_KEY_DOWN:
+    case SDL_EVENT_KEY_UP:
+        return e.key.key;
+    case SDL_EVENT_MOUSE_BUTTON_DOWN:
+    case SDL_EVENT_MOUSE_BUTTON_UP:
+        return (u32)e.button.button;
+    case SDL_EVENT_MOUSE_WHEEL:
+    case SDL_EVENT_MOUSE_WHEEL_OFF:
+        return GetMouseWheelEvent(e);
+    case SDL_EVENT_GAMEPAD_BUTTON_DOWN:
+    case SDL_EVENT_GAMEPAD_BUTTON_UP:
+        return (u32)e.gbutton.button + 0x10000000; // I believe this range is unused
```

Comment:
```suggestion
        return (u32)e.gbutton.button + 0x10000000; // I believe this range is unused
```
What is this magic value? Could we get a constexpr variable with a name instead?

Reply:
It's an offset so the SDL ranges for different types of inputs don't overlap (they all start at 0, and I have to differentiate between them somehow) Keyboard and mouse got the no offset range, as they were the first to be implemented, then for controller buttons, I just chose this as it didn't overlap with anything, and for axes I chose a large offset (0x80000000) so when sorting them they will be at the end of the list, and that is important for reasons. I think a cleaner way to do it would be to finally make this a struct and just give up on storing everything in a single u32 (id, input type, analog value for axes)